### PR TITLE
StorageAuthorizer to Swift

### PR DESCRIPTION
**Pull request by paulb777**
Ports the last FirebaseStorage class and unit tests, StorageAuthorizer to Swift

It depends on the 2.1.0 version of GTMSessionFetcher which recently implemented a new protocol enabling Swift clients to use the authorizer without NSInvocation.  See https://github.com/google/gtm-session-fetcher/pull/315

This PR exposes the only occurrence of the FIRLogger API in Storage.  Swift does not support variadic C function calls so we'll need a better approach.  See https://stackoverflow.com/questions/44246862/call-c-variadic-function-from-swift. Or we may want to do a more fundamental logging redesign.

The combine test changes are the result of more verbose error strings from a previous PR.  I'll note this in the API review.

I plan to do one more cleanup PR in the storage-swift-phase2 before being ready to merge to master.

### CODE REVIEW

File: FirebaseCombineSwift/Tests/Integration/Storage/StorageIntegration.swift [hunk 1]
```diff
@@ -289,7 +289,7 @@ class StorageIntegration: XCTestCase {
         case .finished:
           XCTFail("Unexpected success return from putFile)")
         case let .failure(error):
-          XCTAssertEqual(String(describing: error), "unknown")
+          XCTAssertTrue(String(describing: error).starts(with: "unknown"))
```
Comment (morganchen12) [inline FirebaseCombineSwift/Tests/Integration/Storage/StorageIntegration.swift:292]: For record-keeping purposes, why is this change necessary?

File: FirebaseCore/Extension/FIRLogger.h [hunk 2]
```diff
@@ -116,6 +116,12 @@ extern void FIRLogInfo(FIRLoggerService service, NSString *messageCode, NSString
 extern void FIRLogDebug(FIRLoggerService service, NSString *messageCode, NSString *message, ...)
     NS_FORMAT_FUNCTION(3, 4);
 
+// Swift does not support variadic function calls
+extern void FIRLogDebug1(FIRLoggerService service,
```
Comment (morganchen12) [inline FirebaseCore/Extension/FIRLogger.h:120]: maybe name this `FIRLogDebugSwift`? the `1` is somewhat inscrutable.

File: FirebaseStorage/Sources/Internal/StorageTokenAuthorizer.swift [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+import FirebaseCore
+@_implementationOnly import FirebaseCoreExtension
+
+#if COCOAPODS
+  import GTMSessionFetcher
+#else
+  import GTMSessionFetcherCore
+#endif
+//
+//
```
Comment (morganchen12) [inline FirebaseStorage/Sources/Internal/StorageTokenAuthorizer.swift:28]: empty comments?

File: FirebaseStorage/Sources/Internal/StorageTokenAuthorizer.swift [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+internal class StorageTokenAuthorizer: NSObject, GTMSessionFetcherAuthorizer {
+  func authorizeRequest(_ request: NSMutableURLRequest?,
```
Comment (morganchen12) [inline FirebaseStorage/Sources/Internal/StorageTokenAuthorizer.swift:30]: Should request be non-optional in this method (here and in other methods)?

File: FirebaseStorage/Sources/Internal/StorageTokenAuthorizer.swift [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        }
+        fetchTokenGroup.leave()
+      }
+    }
+    fetchTokenGroup.notify(queue: callbackQueue) {
+      handler(tokenError)
+    }
+  }
+
+  func authorizeRequest(_ request: NSMutableURLRequest?, delegate: Any, didFinish sel: Selector) {
```
Comment (morganchen12) [inline FirebaseStorage/Sources/Internal/StorageTokenAuthorizer.swift:84]: Can this method be removed?

File: FirebaseStorage/Tests/Unit/StorageAuthorizerTests.swift [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+import SharedTestUtilities
+import XCTest
+
+class StorageAuthorizerTests: StorageTestHelpers {
+  var appCheckTokenSuccess: FIRAppCheckTokenResultFake?
+  var appCheckTokenError: FIRAppCheckTokenResultFake?
+  var fetcher: GTMSessionFetcher?
+  var fetcherService: GTMSessionFetcherService?
+  var auth: FIRAuthInteropFake?
+  var appCheck: FIRAppCheckFake?
```
Comment (morganchen12) [inline FirebaseStorage/Tests/Unit/StorageAuthorizerTests.swift:27]: nit: More of a stylistic thing, but imo test variables that are initialized in `setUp` should all be defined as implicitly unwrapped optionals (with a `!` instead of `?`).

**Review by morganchen12 — APPROVED** [review]
LGTM with a couple questions.

File: FirebaseCore/Extension/FIRLogger.h (hunk 2 above)
Comment (paulb777) [reply FirebaseCore/Extension/FIRLogger.h:120]: Done. And found a more general solution. Thanks for the poke.

File: FirebaseStorage/Sources/Internal/StorageTokenAuthorizer.swift (hunk 3 above)
Comment (paulb777) [reply FirebaseStorage/Sources/Internal/StorageTokenAuthorizer.swift:28]: Fixed.

File: FirebaseStorage/Sources/Internal/StorageTokenAuthorizer.swift (hunk 4 above)
Comment (paulb777) [reply FirebaseStorage/Sources/Internal/StorageTokenAuthorizer.swift:30]: It needs to match the protocol.

File: FirebaseStorage/Sources/Internal/StorageTokenAuthorizer.swift (hunk 5 above)
Comment (paulb777) [reply FirebaseStorage/Sources/Internal/StorageTokenAuthorizer.swift:84]: It's needed for the protocol. See the referenced GTMSessionFetcher PR for details.

File: FirebaseStorage/Tests/Unit/StorageAuthorizerTests.swift (hunk 6 above)
Comment (paulb777) [reply FirebaseStorage/Tests/Unit/StorageAuthorizerTests.swift:27]: Done.

**Review by morganchen12 — APPROVED** [review]
LGTM

File: FirebaseCombineSwift/Tests/Integration/Storage/StorageIntegration.swift (hunk 1 above)
Comment (paulb777) [reply FirebaseCombineSwift/Tests/Integration/Storage/StorageIntegration.swift:292]: Updated the description above.
